### ISSUE TITLE
Error in make_imagenet_mean.sh : 'std::length_error'

### ISSUE THREAD

**Issue by drohit92** [BVLC/caffe#1932]
I am trying to follow the ImagNet tutorial.
I had the dataset already, so I just skipped the steps for downloading it.
Next I cropped them to 256*256.
The next step involves computing the mean.

I have the paths all correct in the file: make_imagenet_mean.sh
I am pasting the file contents here: 
./build/tools/compute_image_mean examples/imagenet/ilsvrc12_train_lmdb \ data/ilsvrc12/imagenet_mean.binaryproto
echo "Done."

I run it using:  ./examples/imagenet/make_imagenet_mean.sh

but i get an error:  terminate called after throwing an instance of 'std::length_error'
  what():  basic_string::_S_create
Aborted (core dumped)
Done.

The error suggests that somewhere in the code it's trying to create a string bigger than std::string::max_size().

Can anyone tell me what's the error and how to rectify it?
Thanks

**Comment by drohit92** [comment]
I have tried CIFAR10 example. The compute_image_mean works here. But for the imagenet example it throws the std::length_error. Can someone plz suggest what to do.

Thanks

**Comment by drohit92** [comment]
Ok found the problem. The previous step where I had to crop each image to 256*256, didn't go well.

I realised that the scrypt didn't run entirely, and make_imagenet_mean.sh didn't have access to all images. But I still don't know which particular line caused the error.

Solution:
I was wondering if it's possible to add some code to make a check on size of lmdb files, before the mean is calculated. For e.g train_lmdb is around 240Gb and val_lmdb is ~ 9.5 Gb. This way the make_imagenet_mean.sh should work always and not throw error.
